src/main/java/com/dabsquared/gitlabjenkins/workflow/AcceptGitLabMergeRequestStep.java

```diff
@@ -122,14 +122,13 @@ protected Void run() throws Exception {
         }
 
         private String getCommitMessage(MergeRequest mergeRequest) {
-            if (!step.useMRDescription) return step.mergeCommitMessage;
-
-            String message = "Merge branch '" + mergeRequest.getSourceBranch() + "' into '"
-                    + mergeRequest.getTargetBranch() + "'\n\n" + mergeRequest.getTitle()
-                    + "\n\n" + mergeRequest.getDescription()
-                    + "\n\n" + "See merge request !"
-                    + mergeRequest.getIid();
-            return message;
+            if (!step.useMRDescription || StringUtils.isEmpty(step.mergeCommitMessage)) {
+                return step.mergeCommitMessage;
+            }
```

Comment:
Please explain why you changed the logic to return the empty string if the `mergeCommitMessage` is an empty string.  That seems like a behavior change and I don't see anything in the commit message or the pull request description that explains why the behavior has been changed.
```suggestion
            if (!step.useMRDescription) return step.mergeCommitMessage;
```

Reply:
Yes @MarkEWaite, It was intentional. when we check the `!step.useMRDescription` condition, if this is true then it return the `step.mergeCommitMessage`. And in this case the Commit Message can be empty, so it can lead to a null pointer exception IMO. That's why I added this check. Correct me if I'm wrong.